src/main/java/de/tum/in/www1/artemis/service/programming/ProgrammingLanguageFeatureService.java

```diff
@@ -0,0 +1,54 @@
+package de.tum.in.www1.artemis.service.programming;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+import de.tum.in.www1.artemis.domain.enumeration.ProgrammingLanguage;
+
+/**
+ * This service provides information about features the different ProgrammingLanguages support.
+ * The configuration is also available in the client as the {@link ProgrammingLanguageFeatureContributor} exposes them.
+ */
+@Service
+public class ProgrammingLanguageFeatureService {
+
+    private final Logger log = LoggerFactory.getLogger(ProgrammingLanguageFeatureService.class);
+
+    private final Map<ProgrammingLanguage, ProgrammingLanguageFeature> programmingLanguageFeatures = new HashMap<>();
+
+    public ProgrammingLanguageFeatureService() {
```

Comment:
Sometimes a feature is available using Bamboo but not Jenkins. I don't think this is properly represented in this table. E.g. sequential test runs are only available using Bamboo if I'm not mistaken. You could create different Maps for different VCS+CI configurations.

Reply:
Good point! I changed the `ProgrammingLanguageFeatureService` to an interface and added two implementations (one for Bamboo, one for Jenkins).